dashboard/test/controllers/code_reviews_controller_test.rb

```diff
@@ -0,0 +1,98 @@
+require 'test_helper'
+
+class CodeReviewsControllerTest < ActionController::TestCase
+  self.use_transactional_test_case = true
+
+  setup_all do
+    @project_owner = create :student
+    @storage_id = create_storage_id_for_user(@project_owner.id)
+    @channel_id = create :project, storage_id: @storage_id
+  end
+
+  setup do
+    sign_in @project_owner
+  end
+
+  test 'index returns both open and closed code reviews' do
+    script_id = 12
+    level_id = 5
+
+    closed_at = DateTime.now
+    create :code_review, user_id: @project_owner.id, script_id: script_id, level_id: level_id, closed_at: closed_at
+    create :code_review, user_id: @project_owner.id, script_id: script_id, level_id: level_id, closed_at: closed_at + 1.second
+    create :code_review, user_id: @project_owner.id, script_id: script_id, level_id: level_id, closed_at: nil
+
+    get :index, params: {
+      scriptId: script_id,
+      levelId: level_id
+    }
+    assert_response :success
+
+    response_json = JSON.parse(response.body)
+    assert_equal 3, response_json.length
+  end
+
+  test 'create code review' do
+    script_id = 42
+    level_id = 17
+    project_version = 'abc'
+
+    post :create, params: {
+      scriptId: script_id,
+      levelId: level_id,
+      channelId: @channel_id,
+      version: project_version
+    }
+    assert_response :success
+
+    puts response.body
+    response_json = JSON.parse(response.body)
+    assert_not_nil response_json['id']
+    assert_equal script_id, response_json['scriptId']
+    assert_equal level_id, response_json['levelId']
+    assert_equal project_version, response_json['version']
+    assert_equal false, response_json['isVersionExpired']
+    assert_equal true, response_json['isOpen']
+    assert_not_nil response_json['createdAt']
+  end
+
+  test 'cannot create multiple open code reviews for the same script and level' do
+    script_id = 23
+    level_id = 11
+    project_version = 'abc'
+
+    post :create, params: {
+      scriptId: script_id,
+      levelId: level_id,
+      channelId: @channel_id,
+      version: project_version
+    }
+    assert_response :success
+
+    assert_raises ActiveRecord::RecordInvalid do
```

Comment:
I'm actually a little confused why this raises an exception instead of responding with a 400, but this seemed to be the way we asserted this type of condition elsewhere so I copied it.  :-)

Reply:
It depends a bit on which controller you're looking at :) We have caught these errors and returned a 400 in cases like this, which I think you could do if you wanted. Up to you though.

Reply:
Do you happen to know what the default behavior would be in production?  (It looks like our dev environments are configured to return the full error which I assume wouldn't be the case in production.)

Reply:
Oh, right, it looks this particular exception returns a 422. I tested that locally but can't find any actual documentation on it...

So it sounds like it's (probably) good as is. We probably don't want a 500 here as we don't want it to end up in honeybadger

Reply:
Sounds good, thanks so much for testing this out!